### ISSUE TITLE
Wrong systemd drop-in generated with 17.06

### ISSUE THREAD

**Issue by jpetazzo** [docker/machine#4129]
## What I do

I have an Ubuntu 16.04 machine with docker-ce 17.06-rc2.

I'm using `docker-machine` to "take over" this machine and deploy TLS certs on it, with the following command:
```
docker-machine create -d generic --generic-ssh-user docker --generic-ip-address X.X.X.X
```

## What I expect to see

The Docker Engine starts; my containers run; the people rejoice and celebrates

## What I see instead

The Docker Engine refuses to start.

The file `/etc/systemd/system/docker.service.d/10-machine.conf` contains this:

```
[Service]
ExecStart=
ExecStart=/usr/bin/docker daemon -H tcp://0.0.0.0:2376 -H unix:///var/run/docker.sock --storage-driver aufs --tlsverify --tlscacert /etc/docker/ca.pem --tlscert /etc/docker/server.pem --tlskey /etc/docker/server-key.pem --label provider=generic
Environment=
```

But in 17.06 we should use `dockerd`, not `docker`.

## Extra information

```
ubuntu@node2:~$ docker-machine -v
docker-machine version 0.12.0, build 45c69ad
ubuntu@node2:~$ docker version
Client:
 Version:      17.06.0-ce-rc2
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   402dd4a
 Built:        Wed Jun  7 10:04:47 2017
 OS/Arch:      linux/amd64
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get http://%2Fvar%2Frun%2Fdocker.sock/v1.30/version: dial unix /var/run/docker.sock: connect: permission denied
```

We can't confirm or deny whether this is related to #4114.

**Comment by StefanScherer** [comment]
See #4128 for a fix. The version check in docker-machine through any CE is older than 1.12 and used "docker daemon" instead of "dockerd".

**Comment by dminkovsky** [comment]
I think this can be closed with the release of 0.12.1 @shin-
